### PR TITLE
Add header for client root ca DN (ssl_c_r_dn)

### PR DESCRIPTION
**Pull request by Mrizwanshaik**

### CODE REVIEW

File: jobs/haproxy/templates/haproxy.config.erb [hunk 1]
```diff
@@ -562,9 +562,10 @@ frontend https-in
     http-request set-header X-SSL-Client-Subject-CN %{+Q}[ssl_c_s_dn(cn)]        if { ssl_c_used }
     http-request set-header X-SSL-Client-Issuer-DN  %{+Q}[ssl_c_i_dn]            if { ssl_c_used }
     <%- else %>
-    http-request set-header X-SSL-Client-Subject-DN %{+Q}[ssl_c_s_dn,base64]     if { ssl_c_used }
-    http-request set-header X-SSL-Client-Subject-CN %{+Q}[ssl_c_s_dn(cn),base64] if { ssl_c_used }
-    http-request set-header X-SSL-Client-Issuer-DN  %{+Q}[ssl_c_i_dn,base64]     if { ssl_c_used }
+    http-request set-header X-SSL-Client-Subject-DN %{+Q}[ssl_c_s_dn,base64]            if { ssl_c_used }
+    http-request set-header X-SSL-Client-Subject-CN %{+Q}[ssl_c_s_dn(cn),base64]        if { ssl_c_used }
+    http-request set-header X-SSL-Client-Issuer-DN  %{+Q}[ssl_c_i_dn,base64]            if { ssl_c_used }
+    http-request set-header X-SSL-Client-Root-CA-DN %{+Q}[ssl_c_r_dn,bytes(0,1024)]     if { ssl_c_used }
```
Comment (maxmoehl) [inline jobs/haproxy/templates/haproxy.config.erb:568]: Two thoughts:
* All headers are base64 encoded except this one. We should base64 encode this one as well as there might be characters in the DN which are not supported in headers.
* All headers are unlimited in length except this one. Why is that the case?

File: spec/haproxy/templates/haproxy_config/frontend_https_spec.rb
Comment (maxmoehl) [inline spec/haproxy/templates/haproxy_config/frontend_https_spec.rb:1]: Do we also have acceptance tests for these headers? If so we should add a check for this header there as well.

File: jobs/haproxy/spec [hunk 2]
```diff
@@ -410,6 +410,8 @@ properties:
 
             - X-SSL-Client-Issuer-DN: Contains the base64-encoded issuer distinguished name of the client certificate
 
+            - X-SSL-Client-Root-CA-DN: Contains upto 1024 bytes data of root CA distinguished name of the client certificate
```
Comment (maxmoehl) [inline jobs/haproxy/spec:413]:
```suggestion
            - X-SSL-Client-Root-CA-DN: Contains up to 1024 bytes of the distinguished name of the root CA which signed the client certificate
```

**Review by maxmoehl — CHANGES_REQUESTED** [review]

File: jobs/haproxy/templates/haproxy.config.erb (hunk 1 above)
Comment (Mrizwanshaik) [reply jobs/haproxy/templates/haproxy.config.erb]: - made base64 encoded and removed limit on length. Now we log the complete string.

File: spec/haproxy/templates/haproxy_config/frontend_https_spec.rb
Comment (Mrizwanshaik) [reply spec/haproxy/templates/haproxy_config/frontend_https_spec.rb:1]: There was one in xfcc_test.go, adopted accordingly.

**Review by maxmoehl — CHANGES_REQUESTED** [review]

File: jobs/haproxy/templates/haproxy.config.erb [hunk 3]
```diff
@@ -714,10 +716,12 @@ frontend wss-in
     http-request set-header X-SSL-Client-Subject-DN %{+Q}[ssl_c_s_dn]            if { ssl_c_used }
     http-request set-header X-SSL-Client-Subject-CN %{+Q}[ssl_c_s_dn(cn)]        if { ssl_c_used }
     http-request set-header X-SSL-Client-Issuer-DN  %{+Q}[ssl_c_i_dn]            if { ssl_c_used }
+    http-request set-header X-SSL-Client-Root-CA-DN %{+Q}[ssl_c_r_dn]            if { ssl_c_used }
```
[the remark this answers is not in the document]
Comment (Mrizwanshaik) [reply jobs/haproxy/templates/haproxy.config.erb:725]: Done 👍

**Review by maxmoehl — APPROVED** [review]

**Review by maxmoehl — APPROVED** [review]

**Review by plowin — APPROVED** [review]
for CI

**Pull request merged**